Bindle.Tests/Integration.cs

```diff
@@ -1,19 +1,156 @@
+                    if (!String.IsNullOrEmpty(e.Data))
+                    {
+                        Console.WriteLine(e.Data);
+                    }
+                };
+                process.ErrorDataReceived += (sender, e) =>
+                {
+                    if (!String.IsNullOrEmpty(e.Data))
+                    {
+                        Console.WriteLine(e.Data);
+                    }
+                };
+                process.Start();
+                process.BeginErrorReadLine();
+                process.BeginOutputReadLine();
+                // TODO: something a bit less crappy that this to wait for server to start:
+                Thread.Sleep(5000);
+                Assert.False(process.HasExited);
+
+                // data/invoices/33d7f76b3077de5b72f50d3bf1997d6571770665732ac83b3bede75cf3c5fb67/invoice.toml gets modified by tests so take a copy so it can be reverted at the end.
+
+                origFilePath = Path.Join(dataPath, "invoices/33d7f76b3077de5b72f50d3bf1997d6571770665732ac83b3bede75cf3c5fb67/invoice.toml");
+                copyFilePath = Path.Join(dataPath, "invoices/33d7f76b3077de5b72f50d3bf1997d6571770665732ac83b3bede75cf3c5fb67/invoice.toml.copy");
+                File.Copy(origFilePath, copyFilePath);
+
+            }
+        }
+
+        public void Dispose()
+        {
+
+            // Stop the bindle server
+            try
+            {
+                process?.Kill();
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"Failed to stop bindle process:{ex}");
+            }
+
+            // Delete the files and dirs that the tests created.
+
+            var dirs = new string[]
+            {
+                Path.Join(dataPath, "invoices/8fb420e8308462667f8e1585b7e768bf2e391aec9a29d13c968849c99ed5b0bf"),
+                Path.Join(dataPath, "parcels/460d5965e4d1909e8c7a3748a414956b7038ab5fd79937c9fcb2b214e6b0160a")
+            };
+
+            foreach (var dir in dirs)
+            {
+                if (Directory.Exists(dir))
+                {
+                    try
+                    {
+                        Directory.Delete(dir, true);
+                    }
+                    catch (Exception ex)
+                    {
+                        Console.WriteLine($"Failed to delete test invoice directory: {dir} : {ex}");
+                    }
+
+                }
+            }
+
+            // Revert the modifications
+            
+            if (File.Exists(copyFilePath))
+            {
+                try
+                {
+                    File.Move(copyFilePath, origFilePath, true);
+                }
+                catch (Exception ex)
+                {
+                    Console.WriteLine($"Failed to delete replace modified test invoice:{ex}");
+                }
+            }
+        }
+    }
+
+    // This is a quick and dirty way of ensuring that CanFetchInvoice runs before CanYankInvoice
```

Comment:
Let's do this in a less fragile way, e.g.

- define a `TestOrderAttribute` method attribute with an integer value
- get the attribute value for each method (defaulting to 0 for methods without the attribute)
- order tests by attribute value (where the order between tests with the same value is unspecified)

Then we can apply `[TestOrder(10)]` to `CanYankInvoice` and all should be well.

I know this is a little more work now, but not a huge amount given you already have the `MethodInfo` objects, and makes the order dependency clear and maintainable instead of implicit in the alphabetical order of the names.

Reply:
OK, sounds like a better approach, I will update.